lib/SILOptimizer/Mandatory/MoveOnlyAddressCheckerUtils.cpp

```diff
@@ -2629,6 +2633,80 @@ void MoveOnlyAddressCheckerPImpl::rewriteUses(
 #endif
 }
 
+void MoveOnlyAddressCheckerPImpl::checkForReinitAfterDiscard() {
+  auto const &dropDeinits = addressUseState.dropDeinitInsts;
+  auto const &reinits = addressUseState.reinitInsts;
+
+  if (dropDeinits.empty() || reinits.empty())
+    return;
+
+  using BasicBlockMap = llvm::DenseMap<SILBasicBlock *,
```

Comment:
Nit: I am fine with you not doing this in this patch. But for future reference you can use a Small FrozenMultiMap in situations like this. It is less heavy weight than an llvm DenseMap (which always heap allocates 64 entries). A small frozen multi-map is an array that you initialize with key value pairs that is when it is frozen stable sorted by the first element. So you get a map from Key -> [Value] and it can be in small form.

Reply:
@kavon here is a link to FrozenMultiMap for your perusal. https://github.com/apple/swift/blob/37c8ee2258b276e14ad229ece9ee986cdc53bc00/include/swift/Basic/FrozenMultiMap.h#L15